solidity/contracts/TokenStaking.sol

```diff
@@ -193,7 +191,8 @@ contract TokenStaking is Authorizations, StakeDelegatable {
             _extraData
         );
 
-        emit Staked(_from, _operator, beneficiary, authorizer, _value);
+        emit StakeDelegated(_from, _operator);
+        emit OperatorStaked(_operator, beneficiary, authorizer, _value);
```

Comment:
Did we update unit tests of `delegate` function to cover that both events are emitted?

Reply:
We do not have any tests for emitted events there right now and writing them is quite complex I believe. We are having `approveAndCall` and `receiveApproval` and OpenZeppelin's `expectEvent` is not catching them.

Reply:
Can we use `getPastEvents` for verification?
```js
const startBlock = await web3.eth.getBlock("latest").number

// await delegate

const events = stakingContract.getPastEvents("OperatorStaked", {fromBlock: startBlock })

// assert length of events equals 1 and check events[0] details
```

Reply:
Good idea! Please see https://github.com/keep-network/keep-core/pull/1990/commits/6111cc722fae61312e7c20b4cc880697994c99cc.